src/relation.js

```diff
@@ -37,7 +37,7 @@ export default RelationBase.extend({
       }
       this.parentFk = attributes[this.key('foreignKey')];
     } else {
-      this.parentFk = parent.id;
+      this.parentFk = this.foreignKeyTarget ? parent.attributes[this.foreignKeyTarget] : parent.id;
```

Comment:
This is not very self describing.
```js
this.foreignKeyTarget ? parent.attributes[this.foreignKeyTarget] : parent.id
```
To follow Bookshelfs pattern it would be great to have a method for condition checking.
```js
this.isTargetedForeignKey() ? parent.attributes[this.foreignKeyTarget] : parent.id
// - - - - -
isTargetedForeignKey() {
  return (this.foreignKeyTarget!= null);
}
```